tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -276,29 +270,21 @@ private HttpRequestMessage CreateUpstreamRequest(HttpRequest incomingRequest, by
             {
                 IEnumerable<string> values = header.Value;
 
-                if (s_excludedRequestHeaders.Contains(header.Key, StringComparer.OrdinalIgnoreCase))
-                {
-                    continue;
-                }
-
-                try
+                if (!header.Key.StartsWith("x-recording"))
                 {
-                    if (s_contentRequestHeaders.Contains(header.Key, StringComparer.OrdinalIgnoreCase))
+                    if (upstreamRequest.Headers.TryAddWithoutValidation(header.Key, values))
                     {
-                        upstreamRequest.Content.Headers.TryAddWithoutValidation(header.Key, values);
+                        continue;
                     }
-                    else
+
+                    if(!upstreamRequest.Content.Headers.TryAddWithoutValidation(header.Key, values))
                     {
-                        if (!header.Key.StartsWith("x-recording"))
-                        {
-                            upstreamRequest.Headers.TryAddWithoutValidation(header.Key, values);
-                        }
+                        throw new HttpException(
+                            HttpStatusCode.InternalServerError,
```

Comment:
Should this be a bad request instead?

Reply:
It could be. I was thinking about it from

> We should never hit this situation ever, something has gone massively wrong on the server side.

There should be nothing a client can do that puts us in this state, hence why I'm blaming the server (httpinternal) vs client (badrequest).

Reply:
Couldn't a client add an unknown header?

Reply:
Or would custom headers get added here?

Reply:
Nevermind - I guess custom headers would still get added so I think it is fine how you have it.

Reply:
Any unknown custom headers should add just fine to the standard headers dictionary. That's where our `x-recording-Y` go. We should never get to this code. I will just make it a bad request.